src/identity/interaction/email-password/handler/RegistrationHandler.ts

```diff
@@ -212,54 +210,38 @@ export class RegistrationHandler extends HttpHandler {
    * Verifies that all the data combinations make sense.
    */
   private validateInput(parsed: NodeJS.Dict<string>): ParsedInput {
-    const { email, password, confirmPassword, podName, webId, template, createWebId, register, createPod } = parsed;
+    const { email, password, confirmPassword, webId, podName, register, createPod, createWebId, template } = parsed;
 
-    assert(typeof email === 'string' && email.length > 0 && emailRegex.test(email),
-      'A valid e-mail address is required');
+    // Parse email
+    assert(typeof email === 'string' && emailRegex.test(email), 'Please enter a valid e-mail address.');
 
-    const result: ParsedInput = {
+    const validated: ParsedInput = {
       email,
       template,
+      register: Boolean(register) || Boolean(createWebId),
+      createPod: Boolean(createPod) || Boolean(createWebId),
       createWebId: Boolean(createWebId),
-      register: Boolean(register),
-      createPod: Boolean(createPod),
     };
+    assert(validated.register || validated.createPod, 'Please create a WebID or Pod.');
 
-    const validWebId = typeof webId === 'string' && webId.length > 0;
-    if (result.createWebId) {
-      if (validWebId) {
-        throw new Error('A WebID should only be provided when no new one is being created');
-      }
-    } else {
-      if (!validWebId) {
-        throw new Error('A WebID is required if no new one is being created');
-      }
-      result.webId = webId;
+    // Parse WebID
+    if (!validated.createWebId) {
+      assert(typeof webId === 'string' && /^https?:\/\/[^/]+/u.test(webId), 'Please enter a valid WebID.');
+      validated.webId = webId;
     }
 
-    if (result.register) {
-      assertPassword(password, confirmPassword);
-      result.password = password;
-    } else if (typeof password === 'string' && password.length > 0) {
-      throw new Error('A password should only be provided when registering');
+    // Parse Pod name
+    if (validated.createWebId || validated.createPod) {
+      assert(typeof podName === 'string' && podName.length > 0, 'Please specify a Pod name.');
+      validated.podName = podName;
     }
 
-    if (result.createWebId || result.createPod) {
-      assert(typeof podName === 'string' && podName.length > 0,
-        'A pod name is required when creating a pod and/or WebID');
-      result.podName = podName;
-    } else if (typeof podName === 'string' && podName.length > 0) {
-      throw new Error('A pod name should only be provided when creating a pod and/or WebID');
-    }
-
-    if (result.createWebId && !(result.register && result.createPod)) {
-      throw new Error('Creating a WebID is only possible when also registering and creating a pod');
-    }
```

Comment:
It can never happen: https://github.com/solid/community-server/pull/877/files#diff-8f79e1b79e218035f3143a5f2a66727a29b89ab8a9085d0ade763ba1fca87c80R221-R223

We normally have this case in the tests; it now errors saying that you need to specify a pod name.

Reply:
I missed that. This also works indeed.